include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,617 @@
+
+    //! Return the solution's molar cp. Units: J/kmol/K
+    /*!
+     * Asumming an ideal solution,
+     * \f[
+     * \hat{cp} (T, P, \theta) = \sum_k \theta_k \overline{cp}_k(T, \theta)
+     *                         = \sum_k \theta_k cp^o_k(T, \theta)
+     * \f]
+     *
+     * \see MultiSpeciesThermo
+     */
+    virtual doublereal cp_mole() const;
+
+protected:
+    //! Temporary storage for the reference state enthalpies.
+    mutable vector_fp m_h_ref;
+
+    //! Temporary storage for the reference state entropies.
+    mutable vector_fp m_s_ref;
+
+    //! Temporary storage for the reference state heat capacities.
+    mutable vector_fp m_cp_ref;
+
+    //! Temporary storage for the reference state chemical potentials.
+    mutable vector_fp m_mu_ref;
+
+    //! Temporary storage for the coverage-dependent enthalpies.
+    mutable vector_fp m_h_cov;
+
+    //! Temporary storage for the coverage-dependent entropies.
+    mutable vector_fp m_s_cov;
+
+    //! Temporary storage for the coverage-dependent heat capacities.
+    mutable vector_fp m_cp_cov;
+
+    //! Temporary storage for the coverage-dependent chemical potentials.
+    mutable vector_fp m_mu_cov;
+
+    //! Temporary storage for the sum of reference state enthalpies and
+    //! coverage-dependent enthalpies.
+    mutable vector_fp m_enthalpy;
+
+    //! Temporary storage for the sum of reference state entropies and
+    //! coverage-dependent entropies.
+    mutable vector_fp m_entropy;
+
+    //! Temporary storage for the sum of reference state heat capacities and
+    //! coverage-dependent heat capapcities.
+    mutable vector_fp m_heatcapacity;
+
+    //! Temporary storage for the sum of reference state chemical potentials
+    //! and coverage-dependent chemical potentials.
+    mutable vector_fp m_chempot;
+
+    /*! Indicate the surface coverages have changed by incrementing the
+     *  coverage state number by one. This function is called whenever 
+     *  the setCoverages() or setCoveragesNoNorm() was used to update the
+     *  coverage array.
+     */
+    void coverageChanged();
+
+    //! Array of enthalpy and entropy coverage dependency parameters used in 
+    //! the linear and polynomial models.
+    mutable std::vector<PolynomialDependency> m_PolynomialDependency;
+
+    //! Array of enthalpy and entropy coverage dependency parameters used in 
+    //! the piecewise linear model.
+    mutable std::vector<PiecewiseDependency> m_PiecewiseDependency;
+
+    //! Array of enthalpy and entropy coverage dependency parameters used in 
+    //! the interpolative model.
+    mutable std::vector<InterpolativeDependency> m_InterpolativeDependency;
+
+    //! Array of heat capacity coverage dependency parameters.
+    mutable std::vector<HeatCapacityDependency> m_HeatCapacityDependency;
+
+    //! Storage for the linear and polynomial model coverage dependency 
+    //! parameters for enthalpy.
+    mutable vector_fp m_polynomial_h;
```

Comment:
I think this and a number of the variables below it could just be temporaries declared in the `initThermo` method -- no reason for them to be class members.